### PR TITLE
Fix BatchRequest.execute() multiple window issue

### LINKED ISSUE #10422: MetaMask opens multiple windows for BatchRequest.execute()
**Describe the bug**
When executing batch transactions, MetaMask opens multiple identical approval windows on web3.BatchRequest.execute(), all to approve all of them.

Because an app being able to open as many windows as it wants can be considered a risk, and because all windows opened are identical anyway, only one window should be opened.

Ex.
Metamask screenshot

**Steps to reproduce**

1. Go to https://codesandbox.io/s/004m2
2. Connect your MetaMask wallet
3. Wait 3 seconds after the connect window closes
4. Notice that two identical approval windows have opened for the two transaction batch

**Expected behavior**
Only one window to confirm/reject all the transactions needs to be opened.

**Browser details (please complete the following information):**
 - OS: macOS Mojave 10.14.4
 - Browser: Chrome 87.0.4280.141 (64 bit), Brave 1.18.78 (64 bit), Firefox 63.0.3 (64 bit)
 - MetaMask Version: 8.0.19

**Additional context (Error Messages, etc.)**

This problem is caused by the function **triggerUi** (in **app/scripts/background.js**) being called multiple times on batch execute.

Metamask Broken

Although the function checks whether a window is already open, the window is only considered "open" once the async function to open it (inside) is awaited.

So if **triggerUi** is called several times at once (before any windows are considered open), it will open them all.

Metamask fixed

This problem can be solved by adding a top level variable called **uiIsTriggering** to **background.js**. When **triggerUi** runs, if that variable is true, end the function call with **return;** if it's false, set it to true, then proceed to "try" opening the window. Then, in the try's "finally", set that variable back to false so the window can always be opened again.

### PR DESCRIPTION
**Pull request by jpatel888**
Fixes: #10422

Explanation:  The issue describes the problem with a sandbox, but this PR fixes multiple windows being opened on BatchRequest.execute()

### CODE REVIEW

**Comment by jpatel888** [comment]
I have read the CLA Document and I hereby sign the CLA

File: app/scripts/background.js [hunk 1]
```diff
@@ -482,8 +483,15 @@ async function triggerUi() {
...
     tabs[0].extData &&
     tabs[0].extData.indexOf('vivaldi_tab') > -1;
-  if ((isVivaldi || !popupIsOpen) && !currentlyActiveMetamaskTab) {
-    await notificationManager.showPopup();
+  if (!uiIsTriggering && (isVivaldi || !popupIsOpen) && !currentlyActiveMetamaskTab) {
+    uiIsTriggering = true;
+    try {
+      await notificationManager.showPopup();
+    } catch (err) {
+      log.error('Error triggering UI', err);
```
Comment (Gudahtt) [inline app/scripts/background.js:491]: Could you remove this case? I'd rather the error bubble up and give a more complete stack trace if this fails.

```suggestion
```

**Review by Gudahtt — CHANGES_REQUESTED** [review]
Thanks for the contribution! I noticed this problem recently as well, and had been meaning to create an issue for it.

This solution looks good overall! I just had one suggested change.

File: app/scripts/background.js (hunk 1 above)
Comment (jpatel888) [reply app/scripts/background.js]: Absolutely :) lmk if any other changes are needed.

**Review by Gudahtt — DISMISSED** [review]
LTGM, thanks!

Edit: Oh, `prettier` is angry about something. LGTM after running `yarn lint:fix` then :sweat_smile:

**Comment by jpatel888** [comment]
> LTGM, thanks!
> 
> Edit: Oh, `prettier` is angry about something. LGTM after running `yarn lint:fix` then 😅

I figured it was too easy for there not to be something lol, just added a commit, lmk if there's anything else.

Thanks

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**